src/promptflow/tests/executor/e2etests/test_batch_timeout.py

```diff
@@ -170,3 +171,52 @@ def test_batch_timeout(self, flow_folder):
         # TODO: Currently, the node status is incomplete.
         # We will assert the correct result after refining the implementation of batch timeout.
         assert len(mem_run_storage._node_runs) == 5, "Node runs are persisted in memory storage."
+
+    @pytest.mark.parametrize(
+        "flow_folder",
+        [
+            ASYNC_SAMPLE_FLOW,
+        ],
+    )
+    def test_async_batch_with_line_timeout(self, flow_folder, dev_connections):
```

Comment:
We may also test the case that includes both async/sync tasks in the async scheduler?

Reply:
After adding the test case "async_tools_with_sync_tools", we can come to these conclusions:
1. The async nodes and sync nodes will both be cancelled successfully.
![1706692749582](https://github.com/microsoft/promptflow/assets/39176492/75e16cf8-6a30-4591-8e17-82658d404a30)

2. The handing of CancelledError in node async_passthrough will cause a new problem:
![27eb2a1a596f9256286d13b33545234](https://github.com/microsoft/promptflow/assets/39176492/9473999c-ce5d-4507-b8f0-1ed4041e5fb1)
This is weird and need further investivation.
Currently we added "if" conditon to stdout to mitigate it. The behavior is like, after timeout, some of the CancelledError handing process output will loss and not demonstrated in stdout.